diff --git a/modules/rmvpe.py b/modules/rmvpe.py
index 8d7811b..441563f 100644
--- a/modules/rmvpe.py
+++ b/modules/rmvpe.py
@@ -548,6 +548,7 @@ class RMVPE:
         cents_pred = self.to_local_average_cents(hidden, thred=thred)
         f0 = 10 * (2 ** (cents_pred / 1200))
         f0[f0 == 10] = 0
+        f0 = f0.astype(np.float32, copy=False)  # MPS has no float64.
         # f0 = np.array([10 * (2 ** (cent_pred / 1200)) if cent_pred else 0 for cent_pred in cents_pred])
         return f0
 
